2023/src/day5/almanac.rs

```diff
@@ -12,12 +14,88 @@ pub enum AmcEntryKind {
   HumidityToLocation,
 }
 
+pub struct Almanac {
+  pub seeds: Vec<u64>,
+  pub amc_entries: Vec<AlmanacEntry>,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct AlmanacPair {
+  pub start: u64,
+  pub end: u64,
+}
+impl AlmanacPair {
+  pub fn to_string(&self) -> String {
```

Comment:
The Rust-y way to do this would be to satisfy the `Display` trait by implementing the corresponding `fmt` function, but it's also a little pickier about how you do things I guess.

Reply:
Would that change how it gets pretty-printed? I want to have this for debugging, but not change how it gets pretty-printed when nested in other structs

Reply:
I guess it depends on how you want to print it and when.  imo `Debug` is for splatting out the whole struct on the screen (i.e. when calling `println!("{my_thing:#?}")`) and `Display` is more for converting it into some kind of more human-digestible format (possibly omitting some info or adding more formatting).

Also, I just this very moment learned that implementing `Display` gives you `.to_string()` ([docs](https://doc.rust-lang.org/std/fmt/trait.Display.html)).  Neat.